### PR TITLE
Add OpenStack support

### PR DESCRIPTION
**Pull request by GooBall**
Moves all OpenStack data into an OpenStack view.

### CODE REVIEW

File: views/view-cloud-openstack.php [hunk 1]
```diff
@@ -53,41 +53,38 @@
...
+            $all_status = Array();
 
-        $state = 'instantiated';
-        if (!array_key_exists($all_nodes[$name]['ip'], $instantiated)) {
-            $state = 'uninstantiated';
+            $state = 'instantiated';
+            if (!array_key_exists($all_nodes[$name]['ip'], $instantiated)) {
+                $state = 'uninstantiated';
+            }
+            $all_status[$state] = 'OpenNebula';
```
Comment (jrha) [inline views/view-cloud-openstack.php:85]: Open~~Nebula~~Stack?

File: views/view-cloud-openstack.php [hunk 2]
```diff
@@ -53,41 +53,38 @@
...
+            $results[$group][$panel][$cluster][$name]['note'] = $all_notes[$name];
+        }
+        if (nagios($name) !== Null) {
+            $results[$group][$panel][$cluster][$name]['nagios'] = nagios($name);
+        };
 
-    if ($panel == "opennebula-hypervisor" && array_key_exists($name, $hvstatus)) {
-        $results[$group][$panel][$cluster][$name]['status'] = $hvstatus[$name];
-    }
+        if ($panel == "opennebula-hypervisor" && array_key_exists($name, $hvstatus)) {
```
Comment (jrha) [inline views/view-cloud-openstack.php:74]: open~~nebula~~stack-hypervisor?

**Comment by jrha** [comment]
Seems like there is a fair bit of shared code between the two new cloud views, could it be shared?

File: views/view-cloud-openstack.php (hunk 1 above)
Comment (GooBall) [inline views/view-cloud-openstack.php:85]: As OpenStack is currently running on OpenNebula VMs this information is correct.

File: views/view-cloud-openstack.php (hunk 2 above)
Comment (GooBall) [inline views/view-cloud-openstack.php:74]: See above.

**Comment by GooBall** [comment]
Yes, most of the code is the same but, as we make more things independent of OpenNebula, the code base will change. Right now the benefit of having the views split outweighs the duplicated code.

File: views/view-cloud-openstack.php (hunk 1 above)
Comment (jrha) [inline views/view-cloud-openstack.php:85]: ok!

File: views/view-cloud-openstack.php (hunk 2 above)
Comment (jrha) [inline views/view-cloud-openstack.php:74]: Sure!

**Pull request merged**